client/landing/gutenboarding/components/mshots-image/index.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface MShotsImageProps {
+	url: string;
+	alt: string;
+	'aria-labelledby': string;
+}
+
+export function mshotsUrl( url: string, count = 0 ): string {
+	const mshotsUrl = 'https://s0.wp.com/mshots/v1/';
+	const mShotsParams = {
+		// viewport size (how much of the source page to capture)
+		vpw: 1200,
+		vph: 3072,
+		// size of the resulting image
+		w: 700,
+		h: 1800,
+	};
+	const mshotsRequest = addQueryArgs( mshotsUrl + encodeURIComponent( url ), {
+		...mShotsParams,
+		// this doesn't seem to work:
+		// requeue: true, // Uncomment this line to force the screenshots to be regenerated
+		count,
+	} );
+	return mshotsRequest;
+}
+
+const MShotsImage = ( {
+	url,
+	'aria-labelledby': labelledby,
+	alt,
+}: MShotsImageProps ): JSX.Element => {
+	const [ count, setCount ] = React.useState( 0 );
+	const [ visible, setVisible ] = useState( false );
+	return (
+		<div className="mshots-image__container">
+			{ ! visible && <div className="mshots-image__loader"></div> }
+			<img
+				className={ classnames( 'mshots-image' ) }
+				style={ { opacity: visible ? 1 : 0 } }
+				alt={ alt }
+				aria-labelledby={ labelledby }
+				src={ mshotsUrl( url, count ) }
+				onLoad={ ( e ) => {
+					// Test against mshots h value
+					if ( e.currentTarget.naturalHeight !== 1800 ) {
+						// Triggers a target.src change
+						setTimeout( () => setCount( count + 1 ), 1000 );
```

Comment:
If I understand correctly, this is waiting a full second between the getting the response back from mshots and triggering the next one. I'm seeing time-to-first-bytes of 600ms or more on both the initial query and the `default.png` request (curiously, this includes when chrome reports the image is served from the disk cache - I guess the server must be taking this long to send back a timestamp? )

Anyway, I'm seeing up to two seconds between successive requests for the same image e.g. 5.35s, to 7.57s and then 9.15s and finally 10.74s to initiate the the final successful request.

We totally shouldn't DDOS mshots, but I think we can bring this down. I'd suggest we throttle to 1s rather than wait that long after we get a result.

On the other hand, I saw it get up to `count=3` and still get the `307` 9 seconds after page load, so this is clearly _not_ the bottleneck in the process (*cough* mshots *cough*), so if you agree let's put it into the followup list rather than this PR.

Reply:
Follow up for this I think, it definitely needs some tweaking, but too low results in a sluggish browser as it tries to load 20 images x 3-5 times.